bindings/wasm/src/wallet.rs

```diff
@@ -38,38 +57,34 @@ pub fn create_wallet(options: String) -> Result<WalletMethodHandler, JsValue> {
         .map_err(|e| e.to_string())?;
 
     Ok(WalletMethodHandler {
-        wallet: Arc::new(Mutex::new(Some(wallet_method_handler))),
+        wallet: Arc::new(RwLock::new(Some(wallet_method_handler))),
```

Comment:
Why use the std version?

Reply:
we also do in nodejs. but the std one is not async which is the main reason

Reply:
Yeah but we are in async fns...

Reply:
.client() isnt, just like in rust

Reply:
```rust
pub async fn get_client(method_handler: &WalletMethodHandler) -> Result<ClientMethodHandler, JsValue> {
```